bluemira/equilibria/optimisation/problem/base.py

```diff
@@ -273,25 +199,61 @@ def update_magnetic_constraints(
                 constraint._args["scale"] = self.scale
 
     def _make_numerical_constraints(
-        self,
+        self, coilset: CoilSet
     ) -> tuple[list[ConstraintT], list[ConstraintT]]:
         """Build the numerical equality and inequality constraint dictionaries."""
         if (constraints := getattr(self, "_constraints", None)) is None:
             return [], []
         equality = []
         inequality = []
         for constraint in constraints:
-            f_constraint = constraint.f_constraint()
+            f = constraint.f_constraint()
+
+            f_c = f.f_constraint
+            df_c = getattr(f, "df_constraint", None)
+
+            if coilset._opt_currents_symmetry_status in {
+                CoilSetSymmetryStatus.FULL,
+                CoilSetSymmetryStatus.PARTIAL,
+            }:
+                # wrap the constraint function
+                def wrapped_f_c(x, f=f):
+                    return f.f_constraint(coilset._opt_currents_expand_mat @ x)
```

Comment:
to preserve any docstrings in f.f_constraint, you'll need to import functools probs
```suggestion
                @functools.wraps(f.f_constraint)
                def wrapped_f_c(x, f=f):
                    return f.f_constraint(coilset._opt_currents_expand_mat @ x)
```